pytket/extensions/cutensornet/backends/cutensornet_backend.py

```diff
@@ -214,108 +154,149 @@ def process_circuits(
         Raises:
             TypeError: If global phase is dependent on a symbolic parameter.
         """
-        circuit_list = list(circuits)
-        if valid_check:
-            self._check_all_circuits(circuit_list)
-        handle_list = []
-        for circuit in circuit_list:
-            state_tnet = tk_to_tensor_network(circuit)
-            state = cq.contract(*state_tnet).flatten()
-            try:  # This constraint (from pytket-Qulacs) seems reasonable?
-                phase = float(circuit.phase)
-                coeff = np.exp(phase * np.pi * 1j)
-                state *= coeff  # type: ignore
-            except TypeError:
-                warning(
-                    "Global phase is dependent on a symbolic parameter, so cannot "
-                    "adjust for phase"
-                )
-            res_qubits = [qb for qb in sorted(circuit.qubits)]
-            handle = ResultHandle(str(uuid4()))
-            self._cache[handle] = {
-                "result": BackendResult(q_bits=res_qubits, state=state)
-            }
-            handle_list.append(handle)
-        return handle_list
+        ...
+
+
+class CuTensorNetStateBackend(_CuTensorNetBaseBackend):
+    """A pytket Backend using ``GeneralState`` to obtain state vectors."""
+
+    _supports_state = True
+
+    def __init__(self) -> None:
+        """Constructs a new cuTensorNet backend object."""
+        super().__init__()
+
+    @property
+    def backend_info(self) -> Optional[BackendInfo]:
+        """Returns information on the backend."""
+        return BackendInfo(
+            name="CuTensorNetStateBackend",
+            architecture=None,
+            device_name="NVIDIA GPU",
+            version=__extension_name__ + "==" + __extension_version__,
+            # The only constraint to the gateset is that they must be unitary matrices
+            # or end-of-circuit measurements. These constraints are already specified
+            # in the required_predicates of the backend. The empty set for gateset is
+            # meant to be interpreted as "all gates".
+            # TODO: list all gates in a programmatic way?
+            gate_set=set(),
+            misc={"characterisation": None},
+        )
 
-    # TODO: this should be optionally parallelised with MPI
-    #  (both wrt Pauli strings and contraction itself).
-    def get_operator_expectation_value(
```

Comment:
Unfortunately, this git diff completely butchers what actually happened with `get_operator_expectation_value`: it was moved (verbatim, except for removals of `self` argument) to the `tensor_network_convert.py` file. See https://github.com/CQCL/pytket-cutensornet/pull/130/files#r1653061092 for details on why.